### PR TITLE
New-CmAZCore review update

### PR DESCRIPTION
**Pull request by deepend-dev**

### CODE REVIEW

File: Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1 [hunk 1]
```diff
@@ -63,12 +65,14 @@ function New-CmAzCore {
             $SettingsObject.service.dependencies = $SettingsObject.service.publish
 
             # Core Monitoring
-            if (!$SettingsObject.actionGroups.name) {
+            if (!$SettingsObject.monitor.actionGroups.name) {
 
                 if (!$SettingsObject.monitorSettings) {
-                    $SettingsObject.monitorSettings = "$Directory/monitor.yml"
+                    Write-Error "Please provide settings for monitoring resources.." -Category NotSpecified -TargetObject $SettingsObject.monitorSettings
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1:71]: You be able to specify a category here, there is examples in the codebase.

File: Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1 [hunk 2]
```diff
@@ -114,8 +121,11 @@ function New-CmAzCore {
             if (!$SettingsObject.keyVaults.name) {
 
                 if (!$SettingsObject.keyvaultSettings) {
-                    $SettingsObject.keyvaultSettings = "$Directory/keyvaults.yml"
+                    Write-Error "Please provide keyvault settings.." -Category NotSpecified -TargetObject $SettingsObject.keyvaultSettings
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1:124]: Same as above.

File: Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1 [hunk 3]
```diff
@@ -136,11 +146,13 @@ function New-CmAzCore {
             if (!$SettingsObject.automation.runbook.CertificateName -or !$SettingsObject.automation.runbook.keyVaultCertificatePasswordSecretName -or !$SettingsObject.automation.dsc.keyVaultCertificatePasswordSecretName -or !$SettingsObject.automation.dsc.CertificateName ) {
 
                 if (!$SettingsObject.automationSettings) {
-                    $SettingsObject.automationSettings = "$Directory/automation.yml"
+                    Write-Error "Please provide automation settings.." -Category NotSpecified -TargetObject $SettingsObject.automationSettings
```
Comment (tm-luke) [inline Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1:149]: Same as above.

File: Cloudmarque.Azure/Private/Utility/Resolve-FilePath.ps1 [hunk 4]
```diff
@@ -0,0 +1,18 @@
+function Resolve-FilePath() {
+
+    param(
+        [parameter(Mandatory = $true)]
+        [string]$NestedFile
+    )
+
+    if ($NestedFile.StartsWith('./')) {
+        $NestedFile.replace('./', "$((Get-CmAzContext).ProjectRoot)/")
```
Comment (tm-luke) [inline Cloudmarque.Azure/Private/Utility/Resolve-FilePath.ps1:9]: Store the project root at the top of the file and re-use.

**Review by tm-luke — CHANGES_REQUESTED** [review]

File: Cloudmarque.Azure/Private/Utility/Resolve-FilePath.ps1 (hunk 4 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Private/Utility/Resolve-FilePath.ps1:9]: you might not wanna run that command till there is relative path provided

File: Cloudmarque.Azure/Private/Utility/Resolve-FilePath.ps1 [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    if ($NestedFile.StartsWith('./')) {
+        $NestedFile.replace('./', "$((Get-Item $ParentFile).DirectoryName)/")
+    }
+    elseif ($NestedFile.StartsWith('.\')) {
+        $NestedFile.replace('.\', "$((Get-Item $ParentFile).DirectoryName)\")
+    }
+    else {
+        $NestedFile
+    }
+
```
[the remark this answers is not in the document]
Comment (deepend-dev) [reply Cloudmarque.Azure/Private/Utility/Resolve-FilePath.ps1]: done

File: Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1 (hunk 2 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1]: done

File: Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1 (hunk 3 above)
Comment (deepend-dev) [reply Cloudmarque.Azure/Public/Core/New-CmAzCore.ps1]: done

**Review by tm-luke — APPROVED** [review]

**Review by tm-luke — APPROVED** [review]

**Pull request merged**